docs/nextflow_run/02_pipeline.md

```diff
@@ -0,0 +1,1133 @@
+
+<details>
+  <summary>File contents</summary>
+
+```console title="results/Hello-output.txt"
+Hello
+```
+
+```console title="results/Bonjour-output.txt"
+Bonjour
+```
+
+```console title="results/Holà-output.txt"
+Holà
+```
+
+</details>
+
+This confirms each greeting in the input file has been processed appropriately.
+
+### 1.3. Find the original outputs and logs
+
+You may have noticed that the console output above referred to only one task directory.
+Does that mean all three calls to `sayHello()` were executed within that one task directory?
+
+Let's have a look inside that `8e/0eb066` task directory:
+
+<details>
+  <summary>Directory contents</summary>
+```console title="8e/0eb066"
+work/8e/0eb066071cdb4123906b7b4ea8b047/
+└── Bonjour-output.txt
+```
+</details>
+
+No! We only find the output corresponding to one of the greetings (as well as the accessory files if we enable display of hidden files).
+
+So what's going on here?
+
+By default, the ANSI logging system writes the status information for all calls to the same process on the same line.
+As a result, it only showed us one of the three task directory paths (`8e/0eb066`) in the console output.
+There are two others that are not listed there.
+
+We can modify the logging behavior to see the full list of process calls by adding the `-ansi-log false` to the command as follows:
+
+```bash
+nextflow run 2a-inputs.nf --input greetings.csv -ansi-log false
+```
+
+This time we see all three process runs and their associated work subdirectories listed in the output.
+
+<details>
+  <summary>Command output</summary>
+
+```console linenums="1"
+N E X T F L O W  ~  version 25.04.3
+Launching `2a-inputs.nf` [pedantic_hamilton] DSL2 - revision: 6bbc42e49f
+[ab/1a8ece] Submitted process > sayHello (1)
+[0d/2cae24] Submitted process > sayHello (2)
+[b5/0df1d6] Submitted process > sayHello (3)
+```
+
+Notice that the way the status is reported is a bit different between the two logging modes.
+In the condensed mode, Nextflow reports whether calls were completed successfully or not.
+In this expanded mode, it only reports that they were submitted.
+
+</details>
+
+This confirms that the `sayHello()` process gets called three times, and a separate task directory is created for each one.
+
+If we look inside each of the task directories listed there, we can verify that each one corresponds to one of the greetings.
+
+<details>
+  <summary>Directory contents</summary>
+
+```console title="ab/1a8ece"
+work/ab/1a8ece307e53f03fce689dde904b64/
+└── Hello-output.txt
+```
+
+```console title="0d/2cae24"
+work/0d/2cae2481a53593bc607077c80c9466/
+└── Bonjour-output.txt
+```
+
+```console title="b5/0df1d6"
+work/b5/0df1d642353269909c2ce23fc2a8fa/
+└── Holà-output.txt
+```
+
+</details>
+
+This confirms that each process call is executed in isolation from all the others.
+That has many advantages, including avoiding collisions if the process produces any intermediate files with non-unique names.
```

Comment:
Kind of. Because intermediate files are linked between task directories, if we have non-unique names, we'll still run into collisions. So even though I agree task isolation helps with avoiding collision, it doesn't really avoid. My suggestion is to shorten the phrase (instead of making it longer to be more precise/correct.

```suggestion
That has many advantages, including avoiding collisions.
```